addon/utils/has-block.js

```diff
@@ -0,0 +1,30 @@
+import Ember from 'ember';
+
+const [major, minor] = Ember.VERSION.split('.');
+const isGlimmer = major > 2 || (major == 2 && minor >= 10); // >= 2.10
+
+let hasBlockSymbol;
+
+if (major == 3 && minor >= 1) {
+  // do nothing since the symbol is not exported
+} else if (isGlimmer) {
+  hasBlockSymbol = Ember.__loader.require('ember-glimmer/component')[
+    'HAS_BLOCK'
+  ];
+} else {
+  hasBlockSymbol = Ember.__loader.require('ember-htmlbars/component')[
+    'HAS_BLOCK'
+  ];
+}
+
+// NOTE: I really don't know how to test this
+export default function hasBlock(emberComponent) {
+  // Since Glimmer moved to TypeScript, we can't get the symbol.
+  // This is a terrible but working way to get the value.
+  if (!hasBlockSymbol) {
+    const regex = /HAS_BLOCK/;
+    hasBlockSymbol = Object.keys(emberComponent).find(key => regex.test(key));
```

Comment:
Might be overly defensive, but I could imagine the HAS_BLOCK key being made non iterable, and thus it would be safer to use `Object.getOwnPropertyNames` here.

Reply:
Good point. I am actually afraid Ember will eventually make it real symbol, but even so we can use `getOwnPropertySymbols()`